aws/table_aws_workspaces_directory.go

```diff
@@ -0,0 +1,324 @@
+			},
+			{
+				Name:        "selfservice_permissions",
+				Description: "The default self-service permissions for WorkSpaces in the directory.",
+				Type:        proto.ColumnType_JSON,
+			},
+			{
+				Name:        "state",
+				Description: "The state of the directory's registration with Amazon WorkSpaces.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "subnet_ids",
+				Description: "The identifiers of the subnets used with the directory.",
+				Type:        proto.ColumnType_JSON,
+			},
+			{
+				Name:        "tenancy",
+				Description: "Specifies whether the directory is dedicated or shared.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "workspace_access_properties",
+				Description: "The devices and operating systems that users can use to access WorkSpaces.",
+				Type:        proto.ColumnType_JSON,
+			},
+			{
+				Name:        "workspace_creation_properties",
+				Description: "The default creation properties for all WorkSpaces in the directory.",
+				Type:        proto.ColumnType_JSON,
+			},
+			{
+				Name:        "workspace_security_group_id",
+				Description: "The identifier of the security group that is assigned to new WorkSpaces.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("WorkspaceSecurityGroupId"),
+			},
+			{
+				Name:        "tags_src",
+				Description: "The list of tags for the directory.",
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     listWorkspacesDirectoriesTags,
+				Transform:   transform.FromValue(),
+			},
+
+			// Steampipe standard columns
+			{
+				Name:        "title",
+				Description: resourceInterfaceDescription("title"),
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("DirectoryName"),
+			},
+			{
+				Name:        "tags",
+				Description: resourceInterfaceDescription("tags"),
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     listWorkspacesDirectoriesTags,
+				Transform:   transform.From(workspaceDirectoryTurbotTags),
+			},
+			{
+				Name:        "akas",
+				Description: resourceInterfaceDescription("akas"),
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getWorkspaceDirectoryArn,
+				Transform:   transform.FromValue().Transform(transform.EnsureStringArray),
+			},
+		}),
+	}
+}
+
+//// LIST FUNCTION
+
+func listWorkspacesDirectories(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+
+	// Create Session
+	svc, err := WorkspacesClient(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error("aws_workspaces_directory.listWorkspacesDirectories", "connection_error", err)
+		return nil, err
+	}
+	if svc == nil {
+		// Unsupported region, return no data
+		return nil, nil
+	}
+
+	// Limiting the results
+	maxLimit := int32(25)
+	if d.QueryContext.Limit != nil {
+		limit := int32(*d.QueryContext.Limit)
+		if limit < maxLimit {
+			if limit < 1 {
+				maxLimit = 1
```

Comment:
@madhushreeray30 we do not need to check for limits less than 1. The SDK already takes care of it.